vignettes/CAclust.Rmd

```diff
@@ -32,204 +34,328 @@ knitr::opts_chunk$set(collapse = TRUE, comment = "#>", results = "hold")
 ```{r CA_dimRed}
-# select top variably expressed genes
-cnts <- as.matrix(logcounts(sce)[top_darmanis,])
 # Correspondence Analysis
-caobj = cacomp(cnts,
-               dims = 50,
-               top = nrow(cnts), # number of genes you want to involove in the biclustering and visualization
+caobj = cacomp(sce[top_genes,],
+               assay = "logcounts",
+               dims = 40,
+               top = nrow(cnts), 
                python = TRUE)
 ```
-'python = TRUE' indicates the SVD in CA would be done with a python function which can accelerate the calculation for large datasets. If you fail to set up the python dependencies for APL package, please set it as FALSE.
+The parameter `python = TRUE` indicates that the singular value decomposition in CA is performed with a pythorch implementation. This can drastically speed up the computation, in particular for large datasets. If you did not install all python dependencies for APL, please set it to `FALSE`.
 
-# CAbiNet Biclustering
+## CAbiNet Biclustering
 
-CAbiNet builds up a shared-nearest-neighbour gene-cell graph based on the dimension reduced CA space. To build up the graph, you need to specify the number of nearest neighours (NNs) of cells and genes, which is up to the paramter 'k' in 'caclust' function. 'k' could be an interger indicating an identical number of nearest cell-cell, gene-gene, cell-gene and gene-cell. 'k' can also be set a vector with different number of NNs for cell-cell, gene-gene, cell-gene and gene-cell, respectively, that is, e.g. 'k = c(20, 10, 30, 30)' (Note: the order of intergers in the vectors matters, see '?caclust').
+CAbiNet builds up a shared-nearest-neighbour (SNN) gene-cell graph based on the dimensionality reduced CA space. To build up the graph, you need to specify the number of nearest neighours (NNs) of cells and genes in the function `caclust` with the parameter `k`. As the cell-gene gaph is made up of 4 sub-graphs (cell-cell, gene-gene, cell-gene and gene-cell graph), there are also 4 different ks that can be tuned. If `k` is set to a single integer, CAbiNet will use the same k for all 4 graphs. If however a vector of 4 integers is supplied, it will use them as k for the cell-cell, gene-gene, cell-gene and gene-cell graph in that exact order. So, for example `k = 30` is equivalent to `k = (30, 30, 30, 30)`, but `k = c(30, 15, 30, 30)` uses a k half as large as for the other graphs for the gene-gene graph. In our experience, setting k for the gene-gene graph to approx. half of the other graphs leads to a good visualization/biclustering. See also `?caclust`.
```

Comment:
"In our experience, setting k for the gene-gene graph to approx. half of the other graphs leads to a good visualization/biclustering." I guess this observation only holds for a data set that has more cells than genes. I prefer to remove this suggestion.